### DISCUSSION TITLE
Touch not working in my platform.io project

### DISCUSSION THREAD

**Discussion by honeybadger2023** [Bodmer/TFT_eSPI#2695] [Q&A - Touch]
Hi, I'm trying to get the ILI9341 with touch working on an ESP32 Dev Espressif-WROOM-32 (38Pins) board.

If I use the tutorial provided for platform.io the touch input is not working. It works with arduino-ide.

Did some tests and it looks like the CS pins of both devices are always high. The display itself works. But the touch does nothing.

Could it be a problem with SPI?

I'm using the latest platform.io on the latest Ubuntu.

**Comment by honeybadger2023** [comment]
It starts to get somewhere. After creating a Main.h file for the Main.cpp file and adding all the function declarations I was able to get the button example running. 

But in my main app it still is not working. It seems to be a problem with the folder structure of the project. 

I have the main.h /main.cpp and a folder called "ui_app". Within the ui_app folder everything related to the display ui is stored. The ui is included via "#include ui_app/tft_main.h" in the main.cpp. 

The ui folder and the main.cpp/.h are located in the src folder.

The tft library itself is in the .pio folder.

The ui include works, but it fails to detect the defines from the user_setup.h file. They remain undefined and the touch extension is not loaded. Therefore all the touch related function are not available.

I'm new to platform.io and visual studio so it may be a noob related problem.

**Comment by honeybadger2023** [comment]
Error says include path not fount in browse path. But TFT_eSPI is within .pio.

**Reply by Bodmer** [reply]
I think this is a problem with the project enviroment rather than the library so I have converted to a Q&A discussion topic.

**Reply by Bodmer** [reply]
I don't use platformio so am unable to help. so topic is marked as "help wanted".

**Reply by Pablo2048** [reply]
> Error says include path not fount in browse path. But TFT_eSPI is within .pio.

You have to follow project folders structure for PIO projects, otherwise you have to modify include path(s) in platformio.ini. As default all of the libraries used need to be in lib/ folder (lib/TFT_eSPI) which is the default search path. Please read the documentation here https://docs.platformio.org/en/latest/projectconf/sections/platformio/options/directory/lib_dir.html

**Comment by honeybadger2023** [comment]
Hi, Thanks for the input. Had a mistake in one file that caused the "include path not found error". But the main problem was that I use a project that messes with SPI in a way that those both libraries (FluidNC and TFT_eSPI) can't coexist on one ESP without huger effort.

**Reply by Bodmer** [reply]
If a single SPI bus is shared by 2 libraries then a single SPI class instance must be used by both libraries to ensure the access semaphores work.

TFT_eSPI provides a function to get the SPI instance being used. This may help if the FluidNC library can be instantiated with a particular SPI instance.
